src/cassandra/cluster.rs

```diff
@@ -161,7 +160,18 @@ impl Cluster {
         unsafe {
             let session = Session(cass_session_new());
             let connect_future = <CassFuture<()>>::build(cass_session_connect(session.0, self.0));
-            connect_future.wait().map(|_| session)
+            connect_future.wait()?;
+            Ok(session)
+        }
+    }
+
+    /// Asynchronously connects to the cassandra cluster
+    pub async fn connect_async(&mut self) -> Result<Session> {
```

Comment:
There's a bunch of duplicate code between `connect` and `connect_async`. Why not rewrite `connect` to call `connect_async`? If `.wait()` is good enough for `connect` as written, it ought to be good enough to apply to the result of `connect_async`.

Reply:
Yeah, this makes a lot of sense!

Reply:
OK, doesn't sound worth the engineering. Let's keep this annoying duplication for now.